lib/govspeak.rb

```diff
@@ -390,23 +381,25 @@ def self.devolved_options
         lines << %(<div class="govuk-accordion__section ">)
         lines << %(<div class="govuk-accordion__section-header">)
         lines << %(<h2 class="govuk-accordion__section-heading">)
-        lines << %(<div class="govuk-accordion__section-button" id="accordion-default-heading-#{index}">#{heading}</div>)
+        lines << %(<span class="govuk-accordion__section-button" id="accordion-#{@accordion_index}-heading-#{@accordion_heading_index}">#{heading}</span>)
         lines << %(</h2>)
         lines << summary
         lines << %(</div>)
-        lines << %(<div id="accordion-default-content-#{index}" class="govuk-accordion__section-content" aria-labelledby="accordion-default-heading-#{index}">)
+        lines << %(<div id="accordion-#{@accordion_index}-content-#{@accordion_heading_index}" class="govuk-accordion__section-content" aria-labelledby="accordion-#{@accordion_index}-heading-#{@accordion_heading_index}">)
         lines << %(<div class='govuk-body'>#{content}</div>)
         lines << %(</div>)
         lines << %(</div>)
-        index += 1
+        @accordion_heading_index += 1
       end
-      %(<div class="govuk-accordion" data-module="govuk-accordion" id="accordion-default">#{lines.join}</div>)
+      @accordion_index += 1
+      %(<div class="govuk-accordion" data-module="govuk-accordion" id="accordion-#{@accordion_index - 1}">#{lines.join}</div>)
```

Comment:
I dislike this `-1` in id definition. Can we maybe start the index at 0, increase it by 1 at the beginning of the method, and not bother about subtracting one?

Reply:
or just storing the current index in a local variable - might be more thread safe too? (i'm still so confused by thread safeness lol)

Reply:
Can we store the current index in a local variable? If yes, go for it. If not, eh, `@` variables of an instanciated object seem to be independent from the variables of another instanciated object, at least in rudimentary tests.